src/adapters/scene_direction.js

```diff
@@ -26,13 +26,35 @@ const createMarker = (lngLat, className = '', options = {}) => {
 
 const createRouteLabel = (route, vehicle, { lngLat, anchor }) => {
   const element = document.createElement('div');
-  element.innerHTML = `
-    <div class="routeLabel-vehicleIcon ${getVehicleIcon(vehicle)}"></div>
+  let html = ``;
```

Comment:
Looks like this function suddenly got more complicated, with a lot of low-level HTML manipulation.
Maybe it's time to use our util `renderStaticReact` to be able to extract that as a clean React component and still use it on the map. This is what we already do for `<PoiPopup>`.
What do you think?

Reply:
Sure. I tried to do it like that, it may be not exactly as you would have done it, but you're free to fix it or make me fix it if you have an idea of best practice to use :)